apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2007,15 +2011,18 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
 
     with :ok <- Queries.set_pending_empty_cache(db_client, device_id, true),
          :ok <- VMQPlugin.disconnect("#{realm}/#{encoded_device_id}", true) do
+      new_state = set_device_disconnected(state, db_client, timestamp)
+
+      MessageTracker.ack_delivery(new_state.message_tracker, message_id)
+      Logger.info("Device disconnected.", tag: "device_disconnected")
```

Comment:
I would use a different message to distinguish between a server-initiated disconnection and a client-initiated one, something like
```elixir
Logger.info("Succesfully forced device disconnection.", tag: "forced_device_disconnection")
```